client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -572,9 +572,11 @@ static String buildEndpoint(String... parts) {
             if (Strings.hasLength(part)) {
                 try {
                     //encode each part (e.g. index, type and id) separately before merging them into the path
-                    URIBuilder uriBuilder = new URIBuilder().setPath(part);
-                    //make sure that "/" in each part are properly encoded too
-                    joiner.add(uriBuilder.build().getRawPath().replaceAll("/", "%2F"));
+                    //we prepend "/" to the path part to make this pate absolute, otherwise there can be issues with
```

Comment:
pate -> path

Can we do something like 
if (path.startsWith("/") == false) {
path = "/" + path
}

Reply:
I don't think so. parts are not supposed to start with '/'. If they do, like a document with id `/id`, that `/` is part of the id and needs to be encoded, which we do manually. That first slash is added otherwise URI does crazy things in some cases, like trying to extract the scheme when an index name contains `:` (e.g. `cluster:index`). That doesn't happen if the url is absolute, meaning it starts with `/`. That is why we add it, yet when retrieving the encoded url we ignore it as we do substring(1). This stuff is extremely tricky, and buggy, and there isn't a clean way to do what we need in any of the available method utils that I know of/tried.